jami-android/app/src/main/java/cx/ring/service/RemoteControl.kt

```diff
@@ -0,0 +1,224 @@
+                    ?: throw IllegalStateException("No account found")
+                accountService.addContact(accountId, contactId)
+                Log.d(tag, "Successfully added contact: $contactId to account: $accountId")
+            } catch (e: Exception) {
+                Log.e(tag, "Failed to add contact: $contactId to account: $accountId", e)
+                throw RemoteException("Failed to add contact: ${e.message}")
+            }
+        }
+
+        override fun isContactExist(contactId: String): Boolean {
+            try {
+                val contacts : Map<String, Contact>? = accountService.currentAccount?.contacts
+                return contacts?.containsKey(contactId) ?: false
+            } catch (e: Exception) {
+                Log.e(tag, "Failed to check contact: $contactId", e)
+                throw RemoteException("Failed to check contact: ${e.message}")
+            }
+        }
+
+        override fun sendTrustRequest(contactId: String) {
+            try {
+                val accountId = accountService.currentAccount?.accountId
+                    ?: throw IllegalStateException("No account found")
+                accountService.sendTrustRequest(accountId, contactId)
+                Log.i(tag, "Trust request sent to: $contactId from account: $accountId")
+            } catch (e: Exception) {
+                Log.e(tag, "Failed to send trust request to: $contactId from account: $accountId", e)
+                throw RemoteException("Failed to send trust request: ${e.message}")
+            }
+        }
+
+        override fun initiateCall(userId: String, callback: IRemoteService.ICallback) {
+            Log.d(tag, "Initiating call to user: $userId")
+            try {
+                val disposable = callService.placeCall(
+                    account = getAccountId(),
+                    conversationUri = null,
+                    numberUri = Uri.fromString(userId),
+                    hasVideo = true
+                )
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({ call ->
+                        Log.d(tag, "Call initiated successfully: $call")
+                        callback.onSuccess()
+                    }, { error ->
+                        Log.e(tag, "Failed to initiate call: ${error.message}", error)
+                        callback.onError(error.message)
+                    })
+
+                compositeDisposable.add(disposable)
+            } catch (e: Exception) {
+                Log.e(tag, "Error initiating call", e)
+                callback.onError(e.message)
+            }
+        }
+
+        override fun hangUpCall() {
+            Log.d(tag, "Hanging up the current call")
+            try {
+                val activeCall = callService.getActiveCall() ?: return
+                callService.hangUp(activeCall.account!!, activeCall.daemonIdString!!)
```

Comment:
Is this safe enough? We're calling it quite often.

Reply:
Never had crashes with it.

Reply:
I assume there are no not-null checks in aidl?

Reply:
No, no not-null checks are available there.